sdks/java/core/src/main/java/org/apache/beam/sdk/fn/data/BeamFnDataInboundObserver.java

```diff
@@ -110,6 +114,10 @@ public void close() throws Exception {
     queue.cancel(CloseException.INSTANCE);
   }
 
+  public AtomicBoolean isConsumingReceivedData() {
```

Comment:
I'd return the bool and make the get call here instead of making the caller use the AtomicBool. Encapsulate the implementation detail!

Reply:
Addressed. Thanks!